ckan/lib/helpers.py

```diff
@@ -800,7 +800,7 @@ def _preprocess_dom_attrs(attrs: dict[str, Any]) -> dict[str, Any]:
 def link_to(label: str, url: str, **attrs: Any) -> Markup:
     attrs = _preprocess_dom_attrs(attrs)
     attrs['href'] = url
-    if label == '' or label is None:
+    if label == '':
```

Comment:
helpers could be passed anything from jinja2 including 'none', can't they?

Reply:
Yes, types do not guarantee the actual type. But the same rules apply to `url`. If we are not checking that `url` is a string, why would we do it for `label`?

Taking into account the fact that `label` is a mandatory argument and comes first, there are only 3 ways to get `None` here:

* `None` is passed intentionally: I'd say it's incorrect usage of the function, but since it has no documentation, passing `None` is completely legit.
* This helper is used in the following manner - `h.link_to(PRODUCE_LABEL(), URL)` and `PRODUCE_LABEL` function returns `None`. Same as above
* It called as `h.link_to(LABEL_VAR, url)` and `LABEL_VAR` is undefined. It probably means that there is a problem with application logic.


More strict behavior sounds like an improvement for all 3 cases above. But I don't mind reverting this line, just to be more compatible with the existing ecosystem. It would be much better to create some kind of linter that checks templates and reports misuse of helpers, before making such changes.